ipfx/bin/run_feature_vector_extraction.py

```diff
@@ -88,22 +114,30 @@ def data_for_specimen_id(specimen_id, sweep_qc_option, ap_window_length=0.005):
         return {"error": {"type": "no_ephys_roi_result", "details": ""}}
 
     nwb_path = lq.get_nwb_path_from_lims(roi_id)
-
-    if len(nwb_path) == 0: # could not find an NWB file
+    if (nwb_path is None) or (len(nwb_path) == 0): # could not find an NWB file
         logging.debug("No NWB file for {:d}".format(specimen_id))
         return {"error": {"type": "no_nwb", "details": ""}}
 
     # Check if NWB has lab notebook information, or if additional hdf5 file is needed
     h5_path = None
-    with h5py.File(nwb_path, "r") as h5:
-        if "general/labnotebook" not in h5:
-            h5_path = lq.get_igorh5_path_from_lims(roi_id)
+    try:
```

Comment:
Getting `h5_path` is a good candidate for a separate function.

Reply:
I’m not sure it makes sense for the `h5_path` part to be split from the NWB file loading preceding it, but that whole thing does probably make sense as a separate function.